renku/api/repository.py

```diff
@@ -104,6 +104,10 @@ def __attrs_post_init__(self):
         #: Create an instance of a Git repository for the given path.
         try:
             self.git = Repo(str(self.path))
+            # initialize submodules
```

Comment:
Wouldn't be better to move it to `with_git` decorator or create a new Click option with callback that can be used only with commands where the submodule init is necessary?

Reply:
I thought about moving it to `with_git` but thought that since the submodules are in most cases an integral part of the project, it should just always be done. It incurs very minimal overhead after the first call.

Reply:
```suggestion
            # FIXME refactor submodule initialisation to avoid side-effects
```

Reply:
Let's keep it for now with a `FIXME` comment.

Reply:
👍